extension/parquet/reader/variant/variant_value.cpp

```diff
@@ -0,0 +1,113 @@
+	D_ASSERT(value_type == VariantValueType::OBJECT);
+	object_children.emplace(key, std::move(val));
+}
+
+void VariantValue::AddItem(VariantValue &&val) {
+	D_ASSERT(value_type == VariantValueType::ARRAY);
+	array_items.push_back(std::move(val));
+}
+
+yyjson_mut_val *VariantValue::ToJSON(ClientContext &context, yyjson_mut_doc *doc) const {
+	switch (value_type) {
+	case VariantValueType::PRIMITIVE: {
+		if (primitive_value.IsNull()) {
+			return yyjson_mut_null(doc);
+		}
+		switch (primitive_value.type().id()) {
+		case LogicalTypeId::BOOLEAN: {
+			if (primitive_value.GetValue<bool>()) {
+				return yyjson_mut_true(doc);
+			} else {
+				return yyjson_mut_false(doc);
+			}
+		}
+		case LogicalTypeId::TINYINT:
+			return yyjson_mut_int(doc, primitive_value.GetValue<int8_t>());
+		case LogicalTypeId::SMALLINT:
+			return yyjson_mut_int(doc, primitive_value.GetValue<int16_t>());
+		case LogicalTypeId::INTEGER:
+			return yyjson_mut_int(doc, primitive_value.GetValue<int32_t>());
+		case LogicalTypeId::BIGINT:
+			return yyjson_mut_int(doc, primitive_value.GetValue<int64_t>());
+		case LogicalTypeId::FLOAT:
+			return yyjson_mut_real(doc, primitive_value.GetValue<float>());
+		case LogicalTypeId::DOUBLE:
+			return yyjson_mut_real(doc, primitive_value.GetValue<double>());
+		case LogicalTypeId::DATE:
+		case LogicalTypeId::TIME:
+		case LogicalTypeId::VARCHAR: {
+			auto value_str = primitive_value.ToString();
+			return yyjson_mut_strncpy(doc, value_str.c_str(), value_str.size());
+		}
+		case LogicalTypeId::TIMESTAMP: {
+			string value_str;
+			if (primitive_value_type == LogicalTypeId::TIMESTAMP_TZ) {
+				timestamp_tz_t micros_tz(primitive_value.GetValue<timestamp_t>());
+				auto value = Value::TIMESTAMPTZ(micros_tz);
+				value_str = value.CastAs(context, LogicalType::VARCHAR).GetValue<string>();
+			} else {
+				value_str = primitive_value.CastAs(context, LogicalType::VARCHAR).GetValue<string>();
+			}
+			return yyjson_mut_strncpy(doc, value_str.c_str(), value_str.size());
+		}
+		case LogicalTypeId::TIMESTAMP_NS: {
+			string value_str;
+			if (primitive_value_type == LogicalTypeId::TIMESTAMP_TZ) {
+				auto nanos_ts = primitive_value.GetValue<timestamp_ns_t>();
+
+				//! Convert the nanos timestamp to a micros timestamp
+				date_t out_date;
+				dtime_t out_time;
+				int32_t out_nanos;
+				Timestamp::Convert(nanos_ts, out_date, out_time, out_nanos);
+				auto micros_ts = Timestamp::FromDatetime(out_date, out_time);
+
+				//! Turn the micros timestamp into a micros_tz timestamp and serialize it
+				timestamp_tz_t micros_tz_ts(micros_ts.value);
+				auto value = Value::TIMESTAMPTZ(micros_tz_ts);
+				value_str = value.CastAs(context, LogicalType::VARCHAR).GetValue<string>();
+
+				//! FIXME: we don't support Timestamp(unit=nanos) with timezone in Parquet, to have equivalent results
+				//! between unshredded and shredded values, we shouldn't support it for unshredded values
+				// if (StringUtil::Contains(value_str, "+")) {
```

Comment:
This was changed (as the comment above this line says) in order to match the result between the unshredded and shredded conversion.
I'd prefer to keep this code commented out (not removed) because it produces the correct value for a "Timestamp Nanos with timezone"

We should probably add a `TIMESTAMPNS_TZ` type to DuckDB to support this properly.